keanu-project/src/main/java/io/improbable/keanu/algorithms/mcmc/Hamiltonian.java

```diff
@@ -92,6 +93,8 @@ public NetworkSamples getPosteriorSamples(final BayesianNetwork bayesNet,
         final Map<VertexId, DoubleTensor> momentumBeforeLeapfrog = new HashMap<>();
 
         double logOfMasterPBeforeLeapfrog = bayesNet.getLogOfMasterP();
+        final List<Double> logProbs = new LinkedList<>();
```

Comment:
Is LinkedList the best choice here?  Seems like the overheads to walk and get etc will be pretty high - Perhaps we should consider an ArrayList?

Reply:
Agreed!